lesson13/src/homework.js

```diff
@@ -0,0 +1,119 @@
+// task 1
+const nestedArray = [[1, 2], [3, [4, 5]], 6]
+function flattenArray(array) {
+  return array.flat(Infinity)
+}
+console.log(flattenArray(nestedArray))
+
+// task 2
+const products = [
+  { id: 1, name: 'Телефон', inStock: true },
+  { id: 2, name: 'Ноутбук', inStock: false },
+  { id: 3, name: 'Часы', inStock: false }
+]
+function isAnyProductInStock(product) {
+  return products.some((product) => product.inStock)
+}
+console.log(isAnyProductInStock(products))
+
+// task 3
+const transactions = [
+  { id: 1, type: 'income', amount: 500 },
+  { id: 2, type: 'expense', amount: 100 },
+  { id: 3, type: 'income', amount: 200 },
+  { id: 4, type: 'expense', amount: 50 }
+]
+function getTotalBalance(transactions) {
+  return transactions.reduce((balance, transaction) => (transaction.type === 'income' ? balance + transaction.amount : balance - transaction.amount), 0)
+}
+console.log(getTotalBalance(transactions))
+
+// task 4
+const user = {
+  name: 'John',
+  email: 'john@example.com',
+  age: 25
+}
+function getObjectKeys(obj) {
+  return Object.keys(obj)
+}
+function getObjectValues(obj) {
+  return Object.values(obj)
+}
```

Comment:
```suggestion
function getObjectKeys(object) {
  return Object.keys(object)
}
function getObjectValues(object) {
  return Object.values(object)
}
```

не сокращай названия параметров